controllers/metal3.io/bmceventsubscription_controller.go

```diff
@@ -0,0 +1,303 @@
+		if _, err := prov.RemoveBMCEventSubscriptionForNode(*subscription); err != nil {
+			return errors.Wrap(err, "failed to remove a subscription")
+		}
+
+		// Remove finalizer to allow deletion
+		subscription.Finalizers = utils.FilterStringFromList(
+			subscription.Finalizers, metal3v1alpha1.BMCEventSubscriptionFinalizer)
+		reqLogger.Info("cleanup is complete, removed finalizer",
+			"remaining", subscription.Finalizers)
+		if err := r.Update(context.Background(), subscription); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *BMCEventSubscriptionReconciler) getProvisioner(request ctrl.Request, host *metal3v1alpha1.BareMetalHost) (prov provisioner.Provisioner, ready bool, err error) {
+	reqLogger := r.Log.WithValues("bmceventsubscription", request.NamespacedName)
+
+	prov, err = r.ProvisionerFactory.NewProvisioner(provisioner.BuildHostDataNoBMC(*host), nil)
+	if err != nil {
+		return prov, ready, errors.Wrap(err, "failed to create provisioner")
+	}
+
+	ready, err = prov.IsReady()
+	if err != nil {
+		return prov, ready, errors.Wrap(err, "failed to check services availability")
+	}
+
+	if !ready {
+		reqLogger.Info("provisioner is not ready", "RequeueAfter:", provisionerNotReadyRetryDelay)
+		return prov, ready, nil
+	}
+
+	return prov, ready, nil
+}
+
+func (r *BMCEventSubscriptionReconciler) secretManager(log logr.Logger) secretutils.SecretManager {
+	return secretutils.NewSecretManager(log, r.Client, r.APIReader)
+}
+
+func (r *BMCEventSubscriptionReconciler) getBMCSecret(request ctrl.Request, host *metal3v1alpha1.BareMetalHost) (*corev1.Secret, error) {
+	reqLogger := r.Log.WithValues("bmceventsubscription", request.NamespacedName)
+	secretManager := r.secretManager(reqLogger)
+	bmcCredsSecret, err := secretManager.ObtainSecret(host.CredentialsKey())
+
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return nil, &ResolveBMCSecretRefError{message: fmt.Sprintf("The BMC secret %s does not exist", host.CredentialsKey())}
+		}
+		return nil, err
+	}
+
+	return bmcCredsSecret, nil
+}
+
+func (r *BMCEventSubscriptionReconciler) getHTTPHeaders(subscription metal3v1alpha1.BMCEventSubscription) ([]map[string]string, error) {
+	headers := []map[string]string{}
+
+	if subscription.Spec.HTTPHeadersRef == nil {
+		return headers, nil
+	}
+
+	secret := &corev1.Secret{}
+	secretKey := types.NamespacedName{
+		Name:      subscription.Spec.HTTPHeadersRef.Name,
+		Namespace: subscription.Spec.HTTPHeadersRef.Namespace,
+	}
+
+	err := r.Get(context.TODO(), secretKey, secret)
+
+	if err != nil {
+		return headers, err
+	}
+
+	for headerName, headerValueBytes := range secret.Data {
+		header := map[string]string{}
+		header[headerName] = string(headerValueBytes)
+		headers = append(headers, header)
+	}
+
+	return headers, err
+}
+
+func (r *BMCEventSubscriptionReconciler) buildAndValidateBMCCredentials(request ctrl.Request, host *metal3v1alpha1.BareMetalHost) (bmcCreds *bmc.Credentials, bmcCredsSecret *corev1.Secret, err error) {
```

Comment:
The fix to just use the provisioner for Ironic APi and not need the BMC info looks good, you should also be able remove this function.

Reply:
:man_facepalming: thanks!